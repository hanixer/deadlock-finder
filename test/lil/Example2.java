package showcase;

/**
 * More complex loop
 */
public class Example2 {
    static void func1() {
        int x = 6;
        int y = 3;
        while (x < 100) {
            x = x + 1;
            while (x > 4) {
                x = x - 1;
            }
        }
    }
}