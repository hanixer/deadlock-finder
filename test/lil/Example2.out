func showcase.Example2.func1(): void
bb1: 
var x: int = 6
var y: int = 3
jump bb2

bb2: 
var t~1: boolean = x<100
var t~2: boolean = !t~1
condJump t~2 bb4 bb3

bb4: 
jump end

bb3: 
x = x+1
jump bb5

bb5: 
var t~3: boolean = x>4
var t~4: boolean = !t~3
condJump t~4 bb7 bb6

bb7: 
jump bb2

bb6: 
x = x-1
jump bb5

end: 
return 
